pkg/cmd/server/origin/master.go

```diff
@@ -727,7 +727,12 @@ func (c *MasterConfig) GetRestStorage() map[unversioned.GroupVersion]map[string]
 	importerDockerClientFn := func() dockerregistry.Client {
 		return dockerregistry.NewClient(20*time.Second, false)
 	}
-	imageStreamImportStorage := imagestreamimport.NewREST(importerFn, imageStreamRegistry, internalImageStreamStorage, imageStorage, c.ImageStreamImportSecretClient(), importTransport, insecureImportTransport, importerDockerClientFn)
+	// Always whitelist the integrated registry
+	allowedRegistries := *c.Options.ImagePolicyConfig.AllowedRegistriesForImport
+	if integratedRegistry, ok := c.RegistryNameFn(); ok {
+		allowedRegistries = append([]configapi.RegistryLocation{{DomainName: integratedRegistry}}, allowedRegistries...)
```

Comment:
Doing this here means you can't get a registry lazily, but have to restart.  It would be better to use the lazy method and check at the point in time you validate.